### PR TITLE
[FLINK-23840][runtime] Improve confusing log / exception message when…

### PR DESCRIPTION
**Pull request by dmvk**
https://issues.apache.org/jira/browse/FLINK-23840

### CODE REVIEW

**Comment by alpinegizmo** [comment]
I'm not familiar with how the code here is organized, so no comment on that. But in terms of the user experience, looks great.

**Comment by alpinegizmo** [comment]
If it's easy to do so, it would be nice to backport this to 1.13.

**Review by zentol — CHANGES_REQUESTED** [review]

**Comment by dmvk** [comment]
@zentol Thanks for the review! I've tried to enhance log messages so user can get a better intuition on why a particular options has been ignored.

@alpinegizmo This should be fairly easy to backport as there were no changes to CheckpointStorageLoader since 1.13 release. I've added 1.13.3 to the list of fix versions.

File: flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointStorageLoader.java [hunk 1]
```diff
@@ -44,6 +44,9 @@
 
     private static final String FILE_SYSTEM_STORAGE_NAME = "filesystem";
 
+    private static final String LEGACY_PRECEDENCE_LOG_MESSAGE =
+            "Legacy state backends can also be used as checkpoint storage and are configured to take precedence for backward-compatibility reasons.";
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointStorageLoader.java:48]:
```suggestion
            "Legacy state backends can also be used as checkpoint storage and take precedence for backward-compatibility reasons.";
```
Saying that they are "configured to do X" may imply that this behavior is configurable.

**Review by zentol — CHANGES_REQUESTED** [review]

**Comment by dmvk** [comment]
@zentol Fixed, PTAL ;)

**Pull request merged**